velox/functions/prestosql/aggregates/benchmarks/DispatchBenchmark.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define XXH_INLINE_ALL
+
+#include <folly/Benchmark.h>
+#include "velox/expression/tests/VectorFuzzer.h"
+#include "velox/external/xxhash.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::aggregate;
+
+namespace {
+
+template <typename T>
+FOLLY_ALWAYS_INLINE int64_t hashInteger(const T& value) {
+  return XXH64_round(0, value);
+}
+
+int64_t computeHashesBulkDispatch(VectorPtr vector, memory::MemoryPool* pool) {
+  SelectivityVector rows(vector->size());
+  PrestoHasher hasher(*vector.get(), rows);
+
+  BufferPtr bufferPtr = AlignedBuffer::allocate<int64_t>(vector->size(), pool);
+  hasher.hash(rows, bufferPtr);
+  auto rawBuffer = bufferPtr->as<int64_t>();
+  int64_t hash = 0;
+  for (int i = 0; i < bufferPtr->size(); i++) {
+    hash += rawBuffer[i];
+  }
+  return hash;
+}
+
+int64_t computeHashesOptimized(VectorPtr arrayVector) {
+  // compute Hash for Array of bigints.
+  SelectivityVector rows(arrayVector->size());
+  DecodedVector decodedVector(*arrayVector, rows);
+
+  auto baseArray = decodedVector.base()->as<ArrayVector>();
+  auto rawOffsets = baseArray->rawOffsets();
+  auto rawSizes = baseArray->rawSizes();
+  auto rawElements =
+      baseArray->elements()->as<FlatVector<int64_t>>()->rawValues();
+
+  int64_t hash = 0;
+  rows.applyToSelected([&](auto row) {
+    const auto start = rawOffsets[row];
+    const auto end = start + rawSizes[row];
+    for (int i = start; i < end; i++) {
+      hash = 31 * hash + hashInteger(rawElements[i]);
+    }
+  });
+
+  return hash;
+}
+
+class DispatchBenchmark : public functions::test::FunctionBenchmarkBase {
```

Comment:
My understanding is that benchmark is still work in progress. To unblock this PR, let's move the benchmark to a separate PR.